bokehjs/src/lib/models/scales/scale.ts

```diff
@@ -40,6 +40,14 @@ export abstract class Scale extends Transform {
 
   abstract v_invert(sxs: Arrayable<number>): Arrayable<number>
 
+  screen_min(): number {
+    return this.target_range.start
+  }
+
+  screen_max(): number {
+    return 0
```

Comment:
Thinking out loud, this may need to be a bit more sophisticated. E.g. if the range is flipped with `start > end` then I think you'd want `target_range.end`. Maybe a better long plain description for "screen_min" is "the screen (target) coordinate corresponding to the min data (source) coordinate in the range"

Reply:
totally agree. lots of edge cases to think about.

Reply:
What do you think about `screen_for_data_min`, `screen_for_data_max` as more descriptive names?